packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -189,13 +205,30 @@ class Hero extends StatefulWidget {
   ///
   /// If none is provided, the destination route's Hero child is shown in-flight
   /// by default.
+  ///
+  /// ## Limitations
+  ///
+  /// If a widget built by [flightShuttleBuilder] takes part in a [Navigator]
+  /// push transition, that widget or its descendants must not have any
+  /// [GlobalKey] that is used in the source Hero's descendant widgets. That is
+  /// because both subtrees will be included in the widget tree during the Hero
+  /// flight animation, and [GlobalKey]s must be unique across the entire widget
+  /// tree.
```

Comment:
i don't understand this. Are you saying the `child` will be included twice in the tree? Because that won't work if the widget is stateful... the whole point of the Hero logic is that it only be included once.